### PR TITLE
Intake sim

### PR DESCRIPTION
**Pull request by vichik123**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    public Command setCenterRollerSpeed(MutableMeasure<Velocity<Angle>> speed) {
+        return runOnce(() -> io.setCenterRollerSpeed(speed));
+    }
+
+    @Override
+    public void periodic() {
+        io.updateInputs();
+        Logger.processInputs(this.getClass().getName(), inputs);
+        intakeLigament.setAngle(inputs.currentAngle.baseUnitMagnitude());
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: Are you sure this should be in base units? Better to do this:
```suggestion
        intakeLigament.setAngle(inputs.currentAngle.in(Units.Degrees));
```

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import edu.wpi.first.units.Velocity;
+import lib.motors.SparkMaxSim;
+import lib.motors.TalonFXSim;
+
+public class IntakeIOSim implements IntakeIO {
+    TalonFXSim angleMotor;
+    SparkMaxSim frontRoller;
+    SparkMaxSim centerRoller;
+    PositionVoltage positionRequest = new PositionVoltage(0);
+    PIDController angleController =
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: The controller inside the TalonFXSim class, no need to make it a member.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import edu.wpi.first.units.Units;
+import edu.wpi.first.units.Velocity;
+import lib.motors.SparkMaxSim;
+import lib.motors.TalonFXSim;
+
+public class IntakeIOSim implements IntakeIO {
+    TalonFXSim angleMotor;
+    SparkMaxSim frontRoller;
+    SparkMaxSim centerRoller;
+    PositionVoltage positionRequest = new PositionVoltage(0);
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: Make these private.

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import static edu.wpi.first.units.Units.Degrees;
+
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+
+public class IntakeConstants {
+    public static final double GEAR_RATIO = 0;
+    public static final double ANGLE_KP = 0;
+    public static final double ANGLE_KI = 0;
+    public static final double ANGLE_KD = 0;
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: Should be of type `LoggedTunableNumber`.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Override
+    public void setRollerSpeed(MutableMeasure<Velocity<Angle>> speed) {
+        frontRoller.set(speed.baseUnitMagnitude());
+    }
+
+    @Override
+    public void setCenterRollerSpeed(MutableMeasure<Velocity<Angle>> speed) {
+        centerRoller.set(speed.baseUnitMagnitude());
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: Again, don't use `baseUnitMagnitude`.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Override
+    public void updateInputs() {
+        inputs.currentAngle.mut_replace((angleMotor.getRotorPosition() * 360), Units.Degrees);
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: Why not just do this:
```suggestion
        inputs.currentAngle.mut_replace(angleMotor.getRotorPosition(), Units.Rotations);
```

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 1 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: done.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 2 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: OK

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 3 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: OK

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 4 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: done.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 5 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: changed

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 6 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: done.

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 1 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: 87e7e5cac0b80010fdf2f71007ae5eb93d563786

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 2 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: 87e7e5cac0b80010fdf2f71007ae5eb93d563786

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 3 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: 87e7e5cac0b80010fdf2f71007ae5eb93d563786

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 4 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: 87e7e5cac0b80010fdf2f71007ae5eb93d563786

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 5 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: 87e7e5cac0b80010fdf2f71007ae5eb93d563786

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 6 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: 87e7e5cac0b80010fdf2f71007ae5eb93d563786

**Comment by katzuv** [comment]
`.\gradlew spotlessApply` por favor

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import edu.wpi.first.units.Units;
+import edu.wpi.first.units.Velocity;
+import lib.motors.SparkMaxSim;
+import lib.motors.TalonFXSim;
+
+public class IntakeIOSim implements IntakeIO {
+    private TalonFXSim angleMotor;
+    private SparkMaxSim frontRoller;
+    private SparkMaxSim centerRoller;
+    private PositionVoltage positionRequest = new PositionVoltage(0);
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]:
```suggestion
    private final TalonFXSim angleMotor;
    private final SparkMaxSim frontRoller;
    private final SparkMaxSim centerRoller;
    private final PositionVoltage positionRequest = new PositionVoltage(0);
```

Comment (katzuv) [inline src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: - "position request" is too general, find a more specific name.
- Should use Motion Magic Expo.
- Consult with @vichik123, but I think we should use closed-loop control in all of this subsystem's motors (velocity for the rollers motors). It's not a lot of additional work but will make our intaking more accurate (and it's bonus points from judges).

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+    private IntakeIOSim(PIDController angleController) {
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: Why not create the `PIDController` object in the constants file? There are different gains for sim and real.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.units.Velocity;
+import lib.motors.SparkMaxSim;
+import lib.motors.TalonFXSim;
+
+public class IntakeIOSim implements IntakeIO {
+    private TalonFXSim angleMotor;
+    private SparkMaxSim frontRoller;
+    private SparkMaxSim centerRoller;
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: The proximity sensor is missing but consult with @vichik123 to add it. If you do, use the WPILib simulated class and see how to update its state.

**Review by katzuv — CHANGES_REQUESTED** [review]
Also, I think you should add two convenience methods for retracting and opening the intake (this is something we'll probably use in command groups later on).

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java [hunk 10]
```diff
@@ -4,15 +4,19 @@
...
 import edu.wpi.first.units.MutableMeasure;
+import lib.webconstants.LoggedTunableNumber;
 
 public class IntakeConstants {
-    public static final double ANGLE_RATIO = 0;
-
+    public static final double GEAR_RATIO = 0;
+    public static final LoggedTunableNumber ANGLE_KP = new LoggedTunableNumber("IntakeSim_kP",0);
+    public static final LoggedTunableNumber ANGLE_KI = new LoggedTunableNumber("IntakeSim_kI",0);
+    public static final LoggedTunableNumber ANGLE_KD = new LoggedTunableNumber("IntakeSim_kD",0);
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: Use `/` instead of `_` to split (just like paths in a file system).

**Review by katzuv — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 7 above)
Comment (vichik123) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: Seems a little excessive.

File: src/main/java/frc/robot/RobotContainer.java [hunk 11]
```diff
@@ -1,26 +1,43 @@
...
+import org.eclipse.jetty.util.thread.TimerScheduler;
+
+import java.util.Timer;
+
+import static frc.robot.subsystems.intake.IntakeConstants.*;
 
 public class RobotContainer {
 
     private static RobotContainer INSTANCE = null;
+    private Intake intake = Intake.getInstance();
```
Comment (vichik123) [inline src/main/java/frc/robot/RobotContainer.java]: Be careful here. The `INSTANCE` inside the intake is null until the robotcontainer constructor is called. I'm not sure it's not, but better to call `getInstance` in the constructor.

File: src/main/java/frc/robot/RobotContainer.java [hunk 12]
```diff
@@ -34,7 +51,10 @@ public static RobotContainer getInstance() {
         return INSTANCE;
     }
 
-    private void configureDefaultCommands() {}
+    private void configureDefaultCommands() {
+        intake.setAngle(Units.Degrees.of(0).mutableCopy());
```
Comment (vichik123) [inline src/main/java/frc/robot/RobotContainer.java]: Remember, `setAngle` returns a command, but what do you do with it?

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 13]
```diff
@@ -45,5 +54,6 @@ public Command setCenterRollerSpeed(MutableMeasure<Velocity<Angle>> speed) {
     public void periodic() {
         io.updateInputs();
         Logger.processInputs(this.getClass().getName(), inputs);
+        intakeLigament.setAngle(inputs.currentAngle.in(Units.Radians));
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: The `setAngle` function for the ligament takes degrees.

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 10 above)
Comment (vichik123) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java:13]: Also, the whole idea of tunable numbers is that you don't need to have different constants for sim and real. That means you can just call them `Intake/kP`.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 9 above)
Comment (vichik123) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: The proximity sensor is in the gripper.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+    private PositionVoltage positionRequest = new PositionVoltage(0);
+
+    public IntakeIOSim(PIDController angleController) {
+        angleMotor = new TalonFXSim(1, IntakeConstants.GEAR_RATIO, 0.3);
+        frontRoller = new SparkMaxSim(1, 1, 0);
+        centerRoller = new SparkMaxSim(1, 1, 0);
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: Oh goodness. This is a perfect recipe for an exception. If the moment of inertia is zero, the rotational acceleration would be the moment (force times radius) divided by zero.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 7 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: The only added work is to run SysId and set the gains in code. I'd say, try to tune it for 30 minutes max, if it doesn't work revert to percent output.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 9 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: Right, I forgot.

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 10 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java:13]: How so? A tunable number can't hold two different values.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 7 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: done. 
also we already have a method for convenient retracting called setAngle() in Intake class

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 8 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: possible.

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 10 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: done.

File: src/main/java/frc/robot/RobotContainer.java (hunk 11 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/RobotContainer.java]: done.

File: src/main/java/frc/robot/RobotContainer.java (hunk 12 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/RobotContainer.java]: right......

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 7 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: `setAngle` is not a convenience method, it's a method used for all angle settings. What I meant is that e.g the open method will not receive parameters but use a constant value.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 8 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: Not sure what to do with this reply…

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 14 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: thanks I am good cook.

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 13 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: OK

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 10 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: done.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 7 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: 423a178e32ada435b677b785a6c731fd229beeee

Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: 423a178e32ada435b677b785a6c731fd229beeee

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 8 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: 423a178e32ada435b677b785a6c731fd229beeee

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 10 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: 423a178e32ada435b677b785a6c731fd229beeee

File: src/main/java/frc/robot/RobotContainer.java (hunk 11 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/RobotContainer.java]: 423a178e32ada435b677b785a6c731fd229beeee

File: src/main/java/frc/robot/RobotContainer.java (hunk 12 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/RobotContainer.java]: 423a178e32ada435b677b785a6c731fd229beeee

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 13 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: 423a178e32ada435b677b785a6c731fd229beeee

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 14 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: 423a178e32ada435b677b785a6c731fd229beeee

File: build.gradle [hunk 15]
```diff
@@ -89,7 +89,7 @@ dependencies {
 
     annotationProcessor "org.littletonrobotics.akit.junction:junction-autolog:3.0.2"
 
-    implementation 'com.github.galaxia5987:common:2024.1.1'
+    implementation 'com.github.galaxia5987:common:a2c3f5c264'
```
Comment (katzuv) [inline build.gradle]: Don't commit that here, this should be updated in a separate PR to `develop`, then merged to here (and all other branches).

File: src/main/java/frc/robot/RobotContainer.java [hunk 16]
```diff
@@ -18,12 +18,14 @@ public class RobotContainer {
     private Intake intake;
     private XboxController xboxController = new XboxController(0);
     private JoystickButton a = new JoystickButton(xboxController, XboxController.Button.kA.value);
+    private JoystickButton b = new JoystickButton(xboxController, XboxController.Button.kB.value);
+    private JoystickButton x = new JoystickButton(xboxController, XboxController.Button.kX.value);
+    private JoystickButton y = new JoystickButton(xboxController, XboxController.Button.kY.value);
```
Comment (katzuv) [inline src/main/java/frc/robot/RobotContainer.java]: 1. I don't think you should commit any code to shared files that isn't crucial. Otherwise, you'll have many merge conflicts when you merge all the subsystems. If you need something for testing, just don't commit it.
2. Just [FYI](https://dictionary.cambridge.org/dictionary/english/fyi), you should use [`CommandXboxController`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/button/CommandXboxController.html) instead. That way you won't need to create buttons by yourself.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java [hunk 17]
```diff
@@ -5,27 +5,31 @@
...
+                        1, IntakeConstants.GEAR_RATIO, 0.003, 360 * IntakeConstants.GEAR_RATIO);
         frontRoller = new SparkMaxSim(1, 1, 0.5, 1);
         centerRoller = new SparkMaxSim(1, 1, 0.5, 1);
         angleMotor.setController(IntakeConstants.angleController);
     }
 
     @Override
     public void setAngle(MutableMeasure<Angle> angle) {
+        inputs.angleSetPoint = angle;
+        inputs.currentAngle = Units.Degrees.of(75).mutableCopy();
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: This is probably not true.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java [hunk 18]
```diff
@@ -5,27 +5,31 @@
...
         angleMotor.setControl(positionRequest.withPosition(angle.in(Units.Degrees)));
+        System.out.println(inputs.currentAngle);
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: There is no need for this. I know it's tempting to just spew print statements for debugging, but you don't need that -- all inputs are sent over NetworkTables, and you can watch them using AdvantageScope. This method is faster (no need to redeploy after adding each print) and will also yield better results -- AdvantageScope provides richer data then what you see printed in the console.

**Review by katzuv — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 7 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: And what about the other things?

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 19]
```diff
@@ -60,7 +59,8 @@ public Command setCenterRollerSpeed(MutableMeasure<Velocity<Angle>> speed) {
     public void periodic() {
         io.updateInputs();
         Logger.processInputs(this.getClass().getSimpleName(), inputs);
-        Logger.recordOutput("IntakePose", new Pose3d(new Translation3d(0,0,0), new Rotation3d(0,inputs.currentAngle.in(Units.Degrees),0)));
+        Logger.recordOutput("IntakePose", new Pose3d(new Translation3d(0.327,0,0.165), new Rotation3d(0,-inputs.currentAngle.in(Units.Radians),0)));
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: I think it's better to set angle 0 at the folded position. That way the direction will be as AdvantageScope's CCW+. Another possible solution is to modify the intake's pose in the config JSON file.

**Review by katzuv — CHANGES_REQUESTED** [review]

**Review by katzuv — CHANGES_REQUESTED** [review]
Missing [`Command.withName()`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/Command.html#withName(java.lang.String)) decorators everywhere (should be written like normal text and not in `camelCase`).

File: build.gradle [hunk 20]
```diff
@@ -89,7 +89,7 @@ dependencies {
 
     annotationProcessor "org.littletonrobotics.akit.junction:junction-autolog:3.0.2"
 
-    implementation 'com.github.galaxia5987:common:a2c3f5c264'
+    implementation 'com.github.galaxia5987:common:2024.2.1'
```
Comment (katzuv) [inline build.gradle]: Change in `develop`, then rebase this unto `develop`.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 7 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: And what about `setAngle`?

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 10 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: You fixed my comment, but didn't fix @vichik123's comment.

**Review by katzuv — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 15 above)
Comment (AlonSchwierz) [reply build.gradle]: welp, that is to late. but next time I will do so

File: src/main/java/frc/robot/RobotContainer.java (hunk 16 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/RobotContainer.java]: OK.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 17 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: yup......

do you want to donate to charity?

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 18 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: yeah I know, it was temporary.

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 19 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: can you elaborate? I did not understand.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 17 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: OK

File: src/main/java/frc/robot/RobotContainer.java (hunk 16 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/RobotContainer.java]: 5703c7fe03c04fb3a3f942d77b70f0869886b3a4

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 17 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: 5703c7fe03c04fb3a3f942d77b70f0869886b3a4

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 19 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: CCW+ means "counter-clockwise is positive". Currently, the zero position is the open position (when the intake is parallel to the ground), so to match AdvantageScope's CCW+ direction, the folded position needs to be negative. You can instead set the folded position to be the zero position and then the open position will be 90°. That way you won't need to deal with negative numbers.

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java (hunk 17 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java]: Did you look at the commit's changes before posting its hash?
